lib/Xud.ts

```diff
@@ -39,6 +41,16 @@ class Xud {
     this.logger.info('config loaded');
 
     try {
+      // TODO: wait for decryption of existing key or encryption of new key, config option to disable encryption
```

Comment:
I think you should wrap this block with separate method, on `Xud` class, or even better - inside of `NodeKey` class, since you destined it to handle the persistence anyway.

You can create another static method called `get` or something, which receives as param the file path, and perhaps also the file name, and execute the same logic you currently put in this block, making the other two static methods potentially private.

Reply:
It felt a bit unnatural putting this all into `NodeKey`, also this logic will have to be expanded somewhat with waiting for a password to be provided, so I put this into its own method in `Xud` where I think it makes more sense as part of the initialization procedure. I figure we will want rpc methods to encrypt and decrypt the key eventually but we can discuss more later.